### ISSUE TITLE
Cancellation Requested response

### ISSUE THREAD

**Issue by ruudw86** [JKorf/Binance.Net#743]
Hi,

when i initially use the normal api everything works perfect. But after i while i keep getting an Exception from the wrapper 'Cancellation Requested'.

This suddenly appeared, there is no Error returned by Binance API but generated somewhere in the wrapper.

i cannot find anyhting about is, do you have idea what can cause this?

`
try
{
                using (var client = new Binance.Net.BinanceClient(Globals.BinanceClientOptions))
                {
                    var list = client.Spot.Market.GetKlines(coin, Globals.CandleInterval, null, null, Globals.NumberOfCandlesToRetrieve, new CancellationToken(false));

}
catch(Exception ex)
{
string error = ex.message;
}`

ex.Message = 'Cancellation Requested'

**Comment by HypsyNZ** [comment]
Are you absolutely certain you aren't requesting cancellation somehow?

The way you are creating token might create ambiguity, it should have a name.

```cs
CancellationTokenSource source = new CancellationTokenSource();
CancellationToken token = source.Token;
               
var list = client.Spot.Market.GetKlines(coin, Globals.CandleInterval, null, null, Globals.NumberOfCandlesToRetrieve, token.IsCancellationRequested);
```

How likely are you to request cancellation of this? consider removing it?

**Comment by ruudw86** [comment]
Hi,

at first i didnt send it at all, i have added it to make sure it is set to false.

strange thing is that it seems to happen only at some PC's.
can it have something to do with cpu or memory?

**Comment by HypsyNZ** [comment]
> i have added it to make sure it is set to false.

I see.

> can it have something to do with cpu or memory?

Possibly.

> This suddenly appeared, there is no Error returned by Binance API but generated somewhere in the wrapper.

You should create a ` logger` and set it to `debug` you can make one for the `client` and the `socket`

See how to create a logger here https://github.com/JKorf/Binance.Net/issues/736

**Comment by ruudw86** [comment]
Hi,

2021-05-19 12:16:12:340 | Binance    | Debug | Client configuration: LogVerbosity: Debug, Writers: 1, Credentials: -, BaseAddress: https://api.binance.com/, Proxy: -, RateLimiters: 0, RateLimitBehaviour: Wait, RequestTimeout: 00:00:10
2021-05-19 12:16:12:340 | Binance    | Debug | [59] Creating request for https://api.binance.com/api/v3/klines
2021-05-19 12:16:12:340 | Binance    | Debug | [59] Sending GET request to https://api.binance.com/api/v3/klines?symbol=FTTBTC&interval=30m&limit=1000 
2021-05-19 12:16:22:939 | Binance    | Warning | [59] Request cancel requested
2021-05-19 12:16:22:945 | Binance    | Debug | Disposing exchange client

**Comment by ruudw86** [comment]
i send 100 api calls.
all the same, except the symbol name.

and suddenly in the middle of the proces it start returning this

**Comment by HypsyNZ** [comment]
> Hi,
> https://api.binance.com/, Proxy: -, RateLimiters: 0, RateLimitBehaviour: Wait, **RequestTimeout: 00:00:10**
> 2021-05-19 12:16**:12**:340 | Binance | Debug | [59] Sending GET request to https://api.binance.com/api/v3/klines?
> 2021-05-19 12:16:**22**:939 | Binance | Warning | [59] Request cancel requested

The request is timing out which is causing it to be cancelled as to what is causing that I don't know.

Maybe you are being rate limited, Some limits changed recently.

**Comment by HypsyNZ** [comment]
You could as a workaround just send the request again in your catch

**Comment by HypsyNZ** [comment]
BinanceClient.SetDefaultOptions(new BinanceClientOptions()
                    {
                        RequestTimeout = new TimeSpan(0, 0, 0, 0, 10000),
                        ReceiveWindow = new TimeSpan(0, 0, 0, 0, 1000),
                        RateLimitingBehaviour = RateLimitingBehaviour.Fail,
                    });

Did you set `RequestTimeout ` instead of `ReceiveWindow `

**Comment by ruudw86** [comment]
i have set both to 40 seconds now

2021-05-19 12:27:59:354 | Binance    | Debug | Client configuration: LogVerbosity: Debug, Writers: 1, Credentials: -, BaseAddress: https://api.binance.com/, Proxy: -, RateLimiters: 0, RateLimitBehaviour: Wait, RequestTimeout: 00:00:40
2021-05-19 12:27:59:357 | Binance    | Debug | [85] Creating request for https://api.binance.com/api/v3/klines
2021-05-19 12:27:59:357 | Binance    | Debug | [85] Sending GET request to https://api.binance.com/api/v3/klines?symbol=KSMBTC&interval=30m&limit=1000 
2021-05-19 12:28:01:256 | Binance    | Debug | [85] Response received in 1360ms: [[
2021-05-19 12:28:01:370 | Binance    | Debug | Disposing exchange client
2021-05-19 12:28:03:332 | Binance    | Debug | Client configuration: LogVerbosity: Debug, Writers: 1, Credentials: -, BaseAddress: https://api.binance.com/, Proxy: -, RateLimiters: 0, RateLimitBehaviour: Wait, RequestTimeout: 00:00:40
2021-05-19 12:28:03:332 | Binance    | Debug | [87] Creating request for https://api.binance.com/api/v3/klines
2021-05-19 12:28:03:332 | Binance    | Debug | [87] Sending GET request to https://api.binance.com/api/v3/klines?symbol=LINKBTC&interval=30m&limit=1000 
2021-05-19 12:28:04:067 | Binance    | Debug | [87] Response received in 277ms: [[16
2021-05-19 12:28:04:106 | Binance    | Debug | Disposing exchange client
2021-05-19 12:28:19:179 | Binance    | Debug | Client configuration: LogVerbosity: Debug, Writers: 1, Credentials: -, BaseAddress: https://api.binance.com/, Proxy: -, RateLimiters: 0, RateLimitBehaviour: Wait, RequestTimeout: 00:00:40
2021-05-19 12:28:19:179 | Binance    | Debug | [89] Creating request for https://api.binance.com/api/v3/klines
2021-05-19 12:28:19:179 | Binance    | Debug | [89] Sending GET request to https://api.binance.com/api/v3/klines?symbol=LRCBTC&interval=30m&limit=1000 
2021-05-19 12:28:52:692 | Binance    | Warning | [89] Request exception: An error occurred while sending the request.
  The request was aborted: The request was canceled.

2021-05-19 12:28:52:719 | Binance    | Debug | Disposing exchange client
2021-05-19 12:28:52:719 | Binance    | Debug | Client configuration: LogVerbosity: Debug, Writers: 1, Credentials: -, BaseAddress: https://api.binance.com/, Proxy: -, RateLimiters: 0, RateLimitBehaviour: Wait, RequestTimeout: 00:00:40
2021-05-19 12:28:52:719 | Binance    | Debug | [90] Creating request for https://api.binance.com/api/v3/klines
2021-05-19 12:28:52:719 | Binance    | Debug | [90] Sending GET request to https://api.binance.com/api/v3/klines?symbol=LTCBTC&interval=30m&limit=1000 
2021-05-19 12:29:23:368 | Binance    | Debug | [90] Response received in 18806ms: [[1
2021-05-19 12:29:23:437 | Binance    | Debug | Disposing exchange client
2021-05-19 12:29:38:628 | Binance    | Debug | Client configuration: LogVerbosity: Debug, Writers: 1, Credentials: -, BaseAddress: https://api.binance.com/, Proxy: -, RateLimiters: 0, RateLimitBehaviour: Wait, RequestTimeout: 00:00:40
2021-05-19 12:29:38:628 | Binance    | Debug | [92] Creating request for https://api.binance.com/api/v3/klines
2021-05-19 12:29:38:628 | Binance    | Debug | [92] Sending GET request to https://api.binance.com/api/v3/klines?symbol=LTOBTC&interval=30m&limit=1000 
2021-05-19 12:30:14:511 | Binance    | Warning | [92] Request exception: An error occurred while sending the request.
  The request was aborted: The request was canceled.

**Comment by HypsyNZ** [comment]
This seems like a networking issue but I am not sure what the cause would be

**Comment by JKorf** [comment]
It's been a while, but this was likely to be caused by a request timeout. @ruudw86 can I ask what your environment was with this error (Operating system, dotnet version)? I recently had another developer ask me about this message, which I now believe to be caused by a request timeout. In that case it was on .net framework 4.6 running on windows server, which seems to be working differently than my test environments. 

Long story short; It is a request timeout (30 seconds by default) with an incorrect error message.
